apps/src/pythonlab/pythonHelpers/messageHelpers.ts

```diff
@@ -43,3 +44,32 @@ export function parseErrorMessage(errorMessage: string) {
   const adjustedErrorLines = errorLines.slice(mainErrorLine, errorLines.length);
   return adjustedErrorLines.join('\n');
 }
+
+// This function parses the message string (example: '[PAINTER] PAINT {"color": "Blue"}') to a NeighborhoodSignal.
+export function parseMessageToNeighborhoodSignal(
+  message: string
+): NeighborhoodSignal {
+  /*
+    \[(\w+)]\ captures the signal type inside square brackets, e.g., PAINTER
+    \s+ matches one or more spaces
+    ([^\s]+) captures the value, e.g., PAINT.
+    (\{.*\})? optionally captures the detail json if it exists, e.g., {"color": "Blue"}
+  */
+  const regex = /^\[(\w+)]\s+([^\s]+)(?:\s+(\{.*\}))?$/;
+
+  const match = message.match(regex);
+  if (!match) {
+    throw new Error('Invalid input format');
+  }
+
+  const [, , value, detail] = match;
+
+  const signal = {
```

Comment:
you could write this as 
```
const signal: NeighborhoodSignal = {
...
```
to avoid needing to cast

Reply:
Sounds good. But I think I now need to cast value...
```
  const [, , value, detail] = match;

  const signal: NeighborhoodSignal = {
    value: value as NeighborhoodSignalType,
  };
```

Reply:
that seems fine to me!